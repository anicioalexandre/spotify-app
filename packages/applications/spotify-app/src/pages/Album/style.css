.album-container {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.list-container {
  margin-left: 40px;
  max-height: 65vh;
  overflow-y: auto;
  width: 100%;
}

.image-list-container {
  display: flex;
}

.image-list-container img {
  background: transparent;
  margin-top: 15px;
  margin-right: 40px;
}

.image-container {
  min-height: 500px;
}

@media screen and (max-width: 880px) {
  .image-list-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 40px;
  }
  .list-container {
    margin-left: 0;
  }
  .image-container {
    min-height: unset;
  }
}

.list-container::-webkit-scrollbar {
  width: 8px;
  height: 10px;
}

.list-container::-webkit-scrollbar-track {
  background: none;
}

.list-container::-webkit-scrollbar-thumb {
  background: var(--gray-dark);
}

.list-container::-webkit-scrollbar-thumb:hover {
  background: var(--gray-medium);
}
