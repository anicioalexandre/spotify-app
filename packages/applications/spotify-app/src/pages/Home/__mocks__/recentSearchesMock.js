const recentSearchesMock = {
  last_results: [
    {
      trackId: '3BQHpFgAp4l80e1XslIjNI',
      albumId: '0PT5m6hwPRrpBwIHVnvbFX',
      image: 'https://i.scdn.co/image/ab67616d00001e02e3e3b64cea45265469d4cafa',
      artistName: 'The Beatles',
      name: 'Yesterday - Remastered 2009'
    },
    {
      albumId: '1vANZV20H5B4Fk6yf7Ot9a',
      image: 'https://i.scdn.co/image/ab67616d00001e0255612ece447bec5d62c68375',
      name: 'Beatles For Sale (Remastered)',
      artistName: 'The Beatles'
    },
    {
      albumId: '2zeWAYgh0WfMhiHwvUV15b',
      image: 'https://i.scdn.co/image/ab67616d00001e029f08c2a93ef090a1e1e6bf0c',
      name: 'NÃ³ na Orelha',
      artistName: 'Criolo'
    }
  ]
}

module.exports = { recentSearchesMock }
