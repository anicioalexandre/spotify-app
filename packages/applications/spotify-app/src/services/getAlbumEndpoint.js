import axios from 'axios'

import { SPOTIFY_ALBUMS_API } from './constants'

const getSearchEndpoint = async (token, id) => {
  try {
    const response = await axios.get(SPOTIFY_ALBUMS_API.concat(id), {
      headers: { Authorization: `Bearer ${token}` }
    })
    return Promise.resolve(response.data)
  } catch (error) {
    return Promise.reject(error)
  }
}

export default getSearchEndpoint
