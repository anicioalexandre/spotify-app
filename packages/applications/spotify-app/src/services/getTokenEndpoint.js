import axios from 'axios'
import qs from 'qs'

import { SPOTIFY_TOKEN_API } from './constants'

// const CLIENT_ID = process.env.REACT_APP_CLIENT_ID
// const CLIENT_SECRET = process.env.REACT_APP_CLIENT_SECRET

const CLIENT_ID = '39fc15ee52834938961be68d758d323f'
const CLIENT_SECRET = 'a8ec2a63555f437e9832b5f4e7110763'

const getTokenEndpoint = async () => {
  const headers = {
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/x-www-form-urlencoded'
    },
    auth: {
      username: CLIENT_ID,
      password: CLIENT_SECRET
    }
  }
  const data = {
    grant_type: 'client_credentials'
  }

  try {
    const response = await axios.post(
      SPOTIFY_TOKEN_API,
      qs.stringify(data),
      headers
    )
    return Promise.resolve(response.data.access_token)
  } catch (error) {
    return Promise.reject(error)
  }
}

export default getTokenEndpoint
