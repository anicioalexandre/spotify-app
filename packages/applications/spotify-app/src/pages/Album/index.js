import React, { useEffect } from 'react'
import { useHistory, useParams } from 'react-router'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { getAlbum } from '../../redux/modules/album'
import TracksList from '../../components/TracksList'
import { getAuth } from '../../redux/modules/authToken'
import 'ds-button'
import './style.css'
import LoadingState from '../../components/LoadingState'

const Album = ({ token, getAlbumAction, getAuthAction, album }) => {
  const { id } = useParams()
  const history = useHistory()
  useEffect(() => {
    if (!token) getAuthAction()
    if (token && album.id !== id) getAlbumAction(token, id)
  }, [id, token])
  return (
    <div className="album-container">
      <ds-button onClick={() => history.goBack()} isReturnButton>
        Voltar
      </ds-button>
      <div className="image-list-container">
        <div>
          {album.loading ? (
            <LoadingState width={300} height={300} />
          ) : (
            <ds-image-label
              imageSrc={album.image}
              imageWidth={300}
              imageHeight={300}
              primaryLabel={album.name}
              secondaryLabel={album.artistName}
            />
          )}
        </div>
        <div className="list-container">
          <TracksList dataList={album.tracks} />
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = ({ auth, album }) => {
  return {
    token: auth.token,
    album: {
      artistName: album.artistName,
      id: album.albumId,
      image: album.albumImage,
      name: album.albumName,
      tracks: album.tracks,
      loading: album.loading
    }
  }
}

const mapDispatchToProps = {
  getAuthAction: getAuth,
  getAlbumAction: getAlbum
}

Album.propTypes = {
  token: PropTypes.string,
  getAuthAction: PropTypes.func,
  getAlbumAction: PropTypes.func,
  album: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    id: PropTypes.string,
    image: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    artistName: PropTypes.string.isRequired,
    tracks: PropTypes.arrayOf(
      PropTypes.shape({
        trackDuration: PropTypes.string.isRequired,
        trackId: PropTypes.string.isRequired,
        trackName: PropTypes.string.isRequired
      })
    )
  })
}

export default connect(mapStateToProps, mapDispatchToProps)(Album)
