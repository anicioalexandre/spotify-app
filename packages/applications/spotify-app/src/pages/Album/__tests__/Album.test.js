import React from 'react'
import renderWithRedux from '../../../helpers/renderWithRedux'
import renderWithRouter from '../../../helpers/renderWithRouter'
import Album from '..'
import getTokenEndpoint from '../../../services/getTokenEndpoint'
import getAlbumEndpoint from '../../../services/getAlbumEndpoint'
import { ALBUM_ZE_NETO_MOCK } from '../../../services/__mocks__/getAlbumMock'

afterEach(() => {
  getTokenEndpoint.mockClear()
  getAlbumEndpoint.mockClear()
})

jest.mock('../../../services/getTokenEndpoint')
getTokenEndpoint.mockResolvedValue('token')

jest.mock('../../../services/getAlbumEndpoint')
getAlbumEndpoint.mockResolvedValue(ALBUM_ZE_NETO_MOCK)

describe('Tests for Album page', () => {
  it('renders Album page component', async () => {
    const component = renderWithRedux(renderWithRouter(<Album />))
    await component.findAllByText('Voltar')
    expect(getTokenEndpoint).toHaveBeenCalledTimes(1)
    expect(getAlbumEndpoint).toHaveBeenCalledTimes(1)
    expect(component).toMatchSnapshot()
  })
})
