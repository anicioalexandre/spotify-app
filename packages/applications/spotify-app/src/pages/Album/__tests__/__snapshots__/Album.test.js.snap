// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Album page renders Album page component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="album-container"
      >
        <ds-button
          isreturnbutton="true"
        >
          Voltar
        </ds-button>
        <div
          class="image-list-container"
        >
          <div
            class="image-container"
          >
            <ds-image-label
              imageheight="300"
              imagesrc="https://i.scdn.co/image/ab67616d00001e0229fa460a2fa1f7c16b7c97ae"
              imagewidth="300"
              primarylabel="Zé Neto & Cristiano Voz & Violão"
              secondarylabel="Zé Neto & Cristiano"
            />
          </div>
          <div
            class="list-container"
          >
            <ds-ordered-list
              itemselected=""
            />
          </div>
        </div>
        <div>
          <audio
            class="track-player"
            controls=""
            name="media"
          >
            <source
              src=""
              type="audio/mpeg"
            />
          </audio>
          
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="album-container"
    >
      <ds-button
        isreturnbutton="true"
      >
        Voltar
      </ds-button>
      <div
        class="image-list-container"
      >
        <div
          class="image-container"
        >
          <ds-image-label
            imageheight="300"
            imagesrc="https://i.scdn.co/image/ab67616d00001e0229fa460a2fa1f7c16b7c97ae"
            imagewidth="300"
            primarylabel="Zé Neto & Cristiano Voz & Violão"
            secondarylabel="Zé Neto & Cristiano"
          />
        </div>
        <div
          class="list-container"
        >
          <ds-ordered-list
            itemselected=""
          />
        </div>
      </div>
      <div>
        <audio
          class="track-player"
          controls=""
          name="media"
        >
          <source
            src=""
            type="audio/mpeg"
          />
        </audio>
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
